app/src/main/java/org/wikipedia/descriptions/DescriptionEditFragment.kt

```diff
@@ -226,9 +226,13 @@ class DescriptionEditFragment : Fragment() {
 
         private fun postDescriptionToArticle(editToken: String) {
             val wikiSite = WikiSite.forLanguageCode(pageTitle.wikiSite.languageCode())
-            disposables.add(ServiceFactory.get(wikiSite).getWikiTextForSection(pageTitle.prefixedText, 0)
+            disposables.add(ServiceFactory.get(wikiSite).getWikiTextForSectionWithInfo(pageTitle.prefixedText, 0)
                     .subscribeOn(Schedulers.io())
                     .flatMap { mwQueryResponse ->
+                        if (mwQueryResponse.query()!!.firstPage()!!.getErrorForAction("edit").isNotEmpty()) {
+                            val error = mwQueryResponse.query()!!.firstPage()!!.getErrorForAction("edit")[0]
```

Comment:
Slightly concern about getting the first item from the `getErrorForAction("edit")`, would it be possible to throw an `IndexOutOfBoundsException` if it returns `Collections.emptyList()` and will not reach the `MwException()` below?

Reply:
...but we're already checking if the list is non-empty (above). There will always be at least one item here.

Reply:
Ah! missed that 😅